### ISSUE TITLE
Hang on new tabs after upgrading from 1.38.119 to 1.39.111

### ISSUE THREAD

**Issue by gslin** [brave/brave-browser#23137]
## Description 

Sometimes hang forever (no timeout, I have tried 8 hours) and no content shows on new tabs (e.g. mouserightclick then select "Open link in new tab" or Ctrl-mouseleftclick).

* Disabling all extensions cannot fix this issue.
* Disabling IPv6 cannot fix this issue.
* Using beta channel (1.40.75) cannot fix this issue.
* Downupgrading to 1.38.119 can solve this issue.

## Steps to Reproduce

   1. Use ctrl-leftclick to create many new different tabs.
   2. Wait and see.

## Actual result:

Screencast like this:

https://user-images.githubusercontent.com/7183/170942849-75375dfe-9b19-45a0-8beb-bc13a9aaba9c.mp4

## Expected result:

No hang on new tabs.

## Reproduces how often: 

Intermittent

## Brave version (brave://version info)

On both Linux & Mac environment:
* Ubuntu 20.04 (XFCE, i.e. Xubuntu): Version 1.39.111 Chromium: 102.0.5005.61 (Official Build) (64-bit)
* Mac 12.4: Version 1.39.111 Chromium: 102.0.5005.61 (Official Build) (64-bit) (arm64)

## Version/Channel Information:

- Can you reproduce this issue with the current release? Yes
- Can you reproduce this issue with the beta channel? Yes
- Can you reproduce this issue with the nightly channel? Not tested.

## Other Additional Information:

- Does the issue resolve itself when disabling Brave Shields? No
- Does the issue resolve itself when disabling Brave Rewards? No
- Is the issue reproducible on the latest version of Chrome? No (tested on 95.0.4638.54-1)

**Comment by jaaxx** [comment]
Can confirm on 1.39.111 running on Linux Mint 20.3.  New tabs hang appx 80-90% of the time.  Workaround is to force reload every single tab!

**Comment by jaaxx** [comment]
Update: After some minimal testing it appears the issue may not be present in Beta 1.40.79

**Comment by jaaxx** [comment]
Update: After 24 hours it unfortunately IS still present in Version 1.40.79 Chromium: 102.0.5005.61 (Official Build) beta (64-bit)

**Comment by rebron** [comment]
@jaaxx On issues like these that are intermittent, we need help narrowing down the steps to reproduce what you're seeing. If you can help on specific steps, we maybe able to address the issue here. We'll leave this issue open to see if we're seeing similar reports and possibly more reproducible reports.

**Comment by gslin** [comment]
Okay, I have reproduced this in a clean Amazon EC2 instance.  My guess is this is related to some reactions with XFCE.

My steps to build this EC2 instance:

* Create a `c5a.2xlarge` EC2 instance (AMD-based CPU, since I am running on 5800X).
* Choose Ubuntu 20.04 LTS.
* Install XFCE staging version (i.e. https://launchpad.net/~xubuntu-dev/+archive/ubuntu/staging this) via:
```
sudo add-apt-repository -y ppa:xubuntu-dev/staging
sudo apt update
sudo apt install -y xubuntu-desktop
```
* Install brave-browser:
```
sudo curl -fsSLo /usr/share/keyrings/brave-browser-archive-keyring.gpg https://brave-browser-apt-release.s3.brave.com/brave-browser-archive-keyring.gpg
echo "deb [signed-by=/usr/share/keyrings/brave-browser-archive-keyring.gpg arch=amd64] https://brave-browser-apt-release.s3.brave.com/ stable main"|sudo tee /etc/apt/sources.list.d/brave-browser-release.list
sudo apt update
sudo apt install brave-browser
```
* Install TightVNC server & set up:
```
sudo apt install -y tightvncserver
tightvncserver -geometry 1920x1080 -depth 24
```
* Change `~/.vnc/xstartup` to:
```
#!/bin/sh

xrdb $HOME/.Xresources
xsetroot -solid grey
#x-terminal-emulator -geometry 80x24+10+10 -ls -title "$VNCDESKTOP Desktop" &
#x-window-manager &
# Fix to make GNOME work
#export XKL_XMODMAP_DISABLE=1
#/etc/X11/Xsession
xfce4-session
```
* Reboot this instance.
* Then re-run vnc server:
```
tightvncserver -geometry 1920x1080 -depth 24
```

Now you can use SSH + VNC (like `ssh -L5901:localhost:5901 ubuntu@1.2.3.4` to create SSH tunnel) to access this instance.

Launch Brave, we don't need to adjust any settings to trigger this bug.  But to make us easier to trigger this bug, install https://chrome.google.com/webstore/detail/open-selected-links/hcihcignkpajeehfnomlncinacagapdf this extension, which provides "Open all selected links in the current window" function in the context menu (i.e. mouse rightclick).

Then it's time to trigger this bug with these steps:
1. Open https://blog.gslin.org/ (my blog), select lots of links (e.g. use <kbd>Ctrl</kbd>-<kbd>A</kbd> to select all), then select context menu's "Open all selected links in the current window".
2. Check if all tabs are opened successfully.  In my case they're opened successfully in the first time.
3. Close all tabs.
4. Close the browser.
5. Do the first action again, then you will notice there are some tabs are loading forever.

PS: I could reproduce this bug in my own desktop, even in another new unix account, but I couldn't reproduce in VNC + IceWM, so that's why I went to install XFCE and test again, and bingo...

**Comment by gslin** [comment]
It seems fixed in `Version 1.40.105 Chromium: 103.0.5060.53 (Official Build) (64-bit)` (Linux version).
